executor_manager/common/redis_factory.py

```
@@ -42,80 +46,206 @@ def _get_config(cls) -> RedisConfig:
         Args:
             verify_connection: If True, verify the connection is working
 
         Returns:
             Async Redis client if successful, None if connection failed
         """
+        # Check if existing client is still connected
         if cls._async_client is not None:
-            return cls._async_client
+            try:
+                await cls._async_client.ping()
+                return cls._async_client
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)